gateway/src/main/java/io/zeebe/ResponseMapper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe;
+
+import io.zeebe.broker.client.api.commands.PartitionInfo;
+import io.zeebe.broker.client.api.commands.Topology;
+import io.zeebe.gateway.protocol.GatewayOuterClass.BrokerInfo;
+import io.zeebe.gateway.protocol.GatewayOuterClass.BrokerInfo.Builder;
+import io.zeebe.gateway.protocol.GatewayOuterClass.HealthResponse;
+import io.zeebe.gateway.protocol.GatewayOuterClass.Partition;
+import io.zeebe.gateway.protocol.GatewayOuterClass.Partition.PartitionBrokerRole;
+import java.util.ArrayList;
+
+public class ResponseMapper {
+
+  private PartitionBrokerRole remapPartitionBrokerRoleEnum(
+      final io.zeebe.broker.client.api.commands.PartitionBrokerRole role) {
+    switch (role) {
+      case LEADER:
+        return PartitionBrokerRole.LEADER;
+      case FOLLOWER:
+        return PartitionBrokerRole.FOLLOW;
+    }
+    return null;
```

Comment:
Instead of returning `null` there should be a default case where we throw an exception that the role is unknown. Otherwise the system will throw a `NullPointerException` somewhere else.

Reply:
Which is exactly what we want, since we know that this is going to be used in controller after serialization and we know that controller will have to do error handling (IMHO) I see no problem why this is not perfectly valid to do. 

Now we can talk about it if the client should see error request if the cluster is in an inconsistent state or not. My thinking was that it should, cause every undefined behavior is equal to an error that's why I'm triggering it here.

Now I realize that there is also inconsistency in error types which get handled. If you know a better way of expressing this, please show me.